src/game/game.cpp

```diff
@@ -10298,6 +10307,129 @@ void Game::addPlayerUniqueLogin(std::shared_ptr<Player> player) {
 	m_uniqueLoginPlayerNames[lowercase_name] = player;
 }
 
+bool Game::processChargesOffer(std::shared_ptr<Player> player, uint32_t itemId, uint16_t charges /* = 0*/, bool movable /* = false*/) {
+	std::shared_ptr<Item> newItem = Item::CreateItem(itemId, 1);
+	if (!newItem) {
+		return false;
+	}
+
+	if (charges > 0) {
+		newItem->setAttribute(ItemAttribute_t::CHARGES, charges);
+	}
+
+	if (!movable) {
+		newItem->setAttribute(ItemAttribute_t::STORE, getTimeNow());
+	}
+
+	newItem->setOwner(player);
+
+	std::shared_ptr<Thing> thing = player->getThing(CONST_SLOT_STORE_INBOX);
+	if (!thing) {
+		return false;
+	}
+
+	std::shared_ptr<Item> inboxItem = thing->getItem();
+	if (!inboxItem) {
+		return false;
+	}
+
+	std::shared_ptr<Container> inboxContainer = inboxItem->getContainer();
+	if (!inboxContainer) {
+		return false;
+	}
+
+	auto ret = internalAddItem(inboxContainer, newItem);
+	if (ret != RETURNVALUE_NOERROR) {
+		return false;
+	}
+
+	return true;
+}
+
+bool Game::processStackableOffer(std::shared_ptr<Player> player, uint32_t itemId, uint16_t amount /* = 1*/, bool movable /* = false*/) {
+	std::shared_ptr<Item> newItem = Item::CreateItem(itemId, amount);
+	if (!newItem) {
+		return false;
+	}
+
+	if (!movable) {
+		newItem->setAttribute(ItemAttribute_t::STORE, getTimeNow());
+	}
+
+	newItem->setOwner(player);
+
+	std::shared_ptr<Thing> thing = player->getThing(CONST_SLOT_STORE_INBOX);
+	if (!thing) {
+		return false;
+	}
+
+	std::shared_ptr<Item> inboxItem = thing->getItem();
+	if (!inboxItem) {
+		return false;
+	}
+
+	std::shared_ptr<Container> inboxContainer = inboxItem->getContainer();
+	if (!inboxContainer) {
+		return false;
+	}
+
+	auto ret = internalAddItem(inboxContainer, newItem);
+	if (ret != RETURNVALUE_NOERROR) {
+		return false;
+	}
+
+	return true;
+}
+
+bool Game::processNameChangeOffer(std::shared_ptr<Player> player, std::string &name) {
```

Comment:
Wouldn't it be better to just pass the "name" as a copy? This will avoid having to create a copy inside the function, which should create the same overhead but is more complex.